### DISCUSSION TITLE
Download Previous Version

### DISCUSSION THREAD

**Discussion by lllyasviel** [lllyasviel/Fooocus#1405] [Announcements]
This is a guide for download previous versions. Currently if you want to use a previous version, you need to first get any version of Fooocus running successfully. 

# Windows

### Step 1: Install Fooocus in normal way

Make sure that your Fooocus is running successfully now without other problems.

### Step 2: Stop Update

Edit your `run.bat` (or any other bat launcher), and replace the `entry_with_update` with `launch`. For example,

    .\python_embeded\python.exe -s Fooocus\launch.py
    pause

Note that this will stop all update functionalities forever but will not remove any codes. You can backup or change it back at any time.

### Step 3: Get a previous version (you can skip this if you just want to stop update)

Click the `main` button under the Fooocus title (as shown in the screenshot) to get the list of previous versions, and then select a version that you want. For example 2.1.824

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/8936eea0-ccc6-4598-a646-b41553908c06)

Then click the green or blue `code` button and `download zip`

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/8d07ce7d-dc28-4907-ad82-8fb6b09fc629)

then decompress

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/96d6d8b3-c6eb-4a20-9830-c672afe18379)

then go into the folder `Fooocus-2.1.824`, select all files, then copy and replace all files into your `Fooocus` folder near your `run.bat`.

*Note that the folder `Fooocus-2.1.824` is not a runnable program. It is a patch that you can copy and replace to your Fooocus installation. You need to have at least one installation to use this patch.* 

**Remenber to backup files before replacing files! This operation can not be undo.**

### (if you want to resume and use latest version)

Just download Fooocus again in normal way. The link is [here](https://github.com/lllyasviel/Fooocus/tree/v2.1.824?tab=readme-ov-file#windows).

Note that you can have multiple installations on your device as long as they are in different folder.

If you want to share model directory, see [here](https://github.com/lllyasviel/Fooocus/tree/v2.1.824?tab=readme-ov-file#customization).

# Linux

```
cd Fooocus
```

Pick a version, for example 2.1.824. The command is

```
git pull
git switch v2.1.824
```

Then launch Fooocus with

```
python launch.py
```

To resume update and use latest version:

```
git switch main
git pull
python entry_with_update.py
```

**Comment by rivboyj** [comment]
you seem to be very active on discussions, would you please respond to my post when you can?

**Reply by K3nt3L** [reply]
What is your question ??

**Reply by ilkerd94** [reply]
I tried to download fooocus. I worked until yesterday. Now it says I don't have an NVIDIA GPU. Did something change?

**Reply by K3nt3L** [reply]
Share your GPU'name, if you want some help 🙂

**Comment by amponce** [comment]
Thank you for this incredible project! Fooocus is hands down the best image generation experience that I've come across.

**Comment by FarhanRashid886** [comment]
Can I install the previous version of Fooocus on Mac M1?

**Reply by qazafikamal** [reply]
same question can I use it in Mac with intel iris

**Comment by GodMollo7** [comment]
Are there detailed instructions for the MacBook pro?

**Reply by mary4500** [reply]
Used iPad??

**Comment by GodMollo7** [comment]
There's finally a workaround where it doesn't matter if it's Mac or pc.
I'll get the link off my work phone tomorrow

**Comment by GodMollo7** [comment]
No, click this link in google chrome:
https://colab.research.google.com/github/lllyasviel/Fooocus/blob/main/fooocus_colab.ipynb#scrollTo=VjYy0F2gZIPR
when it's done hit the link & fooocus will load in another window

**Comment by GodMollo7** [comment]
https://youtu.be/SXpoGTQF7fs?si=QkUXI_X7FtKnXKOl

**Comment by GodMollo7** [comment]
Thats the youtube video giving full instructions

**Comment by tektek31** [comment]
run.bat run.bat does not work The path specified was not found

**Comment by ilkerd94** [comment]
I have a lenovo thinbook and I'm really bad with Python and very new to
github. I ran the app a few times through (COLLAB) and it worked fine. Wen
I try to rerun it it says I don't have an nvidia gpu but sometimes I can
use it, sometimes I can not. My discord is ilk1453

Op vr 18 apr 2025 om 21:26 schreef Marc Quentel ***@***.***>:

**Comment by K3nt3L** [comment]
No Nvidia GPU with Lenovo, and AMD is only supported with Linux. You've got the wrong computer for inference.

**Comment by ilkerd94** [comment]
But I can sitll use it 90% of the time. Sometimes it just gives me a
runtime error that I have no driver, but if keep trying enough times, I can
get on the gradio.live site and generate images for a couple hours.

Op za 19 apr 2025 om 14:04 schreef Marc Quentel ***@***.***>:

**Comment by K3nt3L** [comment]
Check this :
https://github.com/microsoft/DirectML?tab=readme-ov-file#hardware-requirements
But first how many Go of VRAM ?

**Comment by ilkerd94** [comment]
[image: image.png]Chip type: Inter (R) Iris (R) Xe Graphics Family
DAC type: Internal
adapter string: Intel (R) Iris (R) Xe Graphics
Bios information: intel video BIOS
Total available GM: 8219MB
dedicated video memory: 128MB
System video memory: 0GB
Shared system memory: 8091

Op vr 18 apr 2025 om 21:26 schreef Marc Quentel ***@***.***>:
